cookbooks/bcpc/recipes/powerdns.rb

```diff
@@ -45,259 +44,214 @@
 subnet = node[:bcpc][:management][:subnet]
 raise "Did not get a subnet" if not subnet
 
-%w{pdns-server pdns-backend-mysql}.each do |pkg|
-    package pkg do
-        action :upgrade
-    end
-end
+if node[:bcpc][:management][:vip] and get_nodes_for("mysql").length() > 0
+#  make_config('mysql-pdns-user', "pdns")
+#  make_config('mysql-pdns-password', secure_password)
+
+  allnodes = Array.new
```

Comment:
I will still have to figure out information for VIP host as there is no node object for it. I still expect some sort of duplicate code in `PowerDNS` recipe.